src/samplers/disk/mod.rs

```diff
@@ -139,36 +141,51 @@ impl Disk {
                     include_str!("../../common/value_to_index2.c"),
                 );
                 let mut bpf = bcc::BPF::new(&code)?;
-                // load + attach kprobes!
-                bcc::Kprobe::new()
-                    .handler("trace_pid_start")
-                    .function("blk_account_io_start")
-                    .attach(&mut bpf)?;
-                if let Ok(results) = bpf.get_kprobe_functions("blk_start_request") {
-                    if !results.is_empty() {
-                        bcc::Kprobe::new()
-                            .handler("trace_req_start")
-                            .function("blk_start_request")
-                            .attach(&mut bpf)?;
+                // collect the set of probes required from the statistics enabled.
+                let mut probes = HashSet::new();
+                for statistic in &self.statistics {
+                    for probe in statistic.bpf_probes_required() {
+                        probes.insert(probe);
                     }
                 }
-                bcc::Kprobe::new()
-                    .handler("trace_req_start")
-                    .function("blk_mq_start_request")
-                    .attach(&mut bpf)?;
-                if let Ok(results) = bpf.get_kprobe_functions("blk_account_io_completion") {
-                    if !results.is_empty() {
-                        bcc::Kprobe::new()
-                            .handler("do_count")
-                            .function("blk_account_io_completion")
-                            .attach(&mut bpf)?;
-                    } else {
-                        bcc::Kprobe::new()
-                            .handler("do_count")
-                            .function("blk_account_io_done")
-                            .attach(&mut bpf)?;
+
+                // load + attach the kernel probes that are required to the bpf instance.
```

Comment:
Can you explain what you're trying to do here? It's not exactly clear to me

Reply:
The original code in dis/mod.rs is different from other samplers. In other samplers, they simply create probes and attach them, and that's why the new implementation here looks a bit different and more complicated as well.

To be more specific, here it does some checks, e.g. it checks if 'blk_start_request' can be found first before attaching them.
It also chooses between 'blk_account_io_completion' and 'blk_account_io_done' on the condition of whether 'blk_account_io_completion' exists or not.

What I am trying to do here is not to change the existing logic/behaviour.